src/som/interpreter/nodes/ExpressionWithReceiverNode.java

```diff
@@ -15,6 +15,8 @@ public ExpressionWithReceiverNode(final SourceSection sourceSection) {
   public Object evaluateReceiver(final VirtualFrame frame){
     return this.getReceiver().executeGeneric(frame);
   }
+
+  public abstract Object executeGenericWithReceiver(final VirtualFrame frame, final Object receiver);
```

Comment:
Ok, so, this is where I think this solution is much less nice then the previous one with the temp variable.
This seems to introduce quite a bit more code, which is specific to one language feature. Not sure this is a good tradeoff.

I am not sure there is any way around introducing such a method though. Hmmm. Yeah, the problem is that this essentially 'breaks' the notion that each node evaluates all its child nodes. hmmm.

Don't know.

Do you by chance have a performance comparison for both implementations on some microbenchmarks using cascaded message sends? Perhaps that helps making a decision?